### PR TITLE
Align GPTQ keep masks with captured tensor devices

### PR DESCRIPTION
**Pull request by peter-positron**
## Summary

GPTQ capture raises an indexing-device error when a calibration keep mask and the captured tensor are on different CUDA devices. Move each sample's mask to the input and output tensor devices independently before indexing.

## What Changed

- Preserve per-sample filtering, empty-sample skipping, and batch indices.
- Add CPU, same-GPU, and two-GPU tests checking captured rows, sample counts, and Hessian contents.

## Tests

- [x] I added a new simple/fast unit test for this change.
- [x] I ran the new targeted test locally before opening this PR.
- [x] I ran the directly relevant local tests.

Python 3.13.13, torch 2.11.0+cu128, Transformers 5.16.1; two CUDA GPUs:

```bash
PYTHONDONTWRITEBYTECODE=1 PYTHON_GIL=1 OMP_NUM_THREADS=1 \
  CUDA_VISIBLE_DEVICES=0,1 python -m pytest -p no:cacheprovider -q \
  tests/test_gptq_keep_mask_device.py \
  tests/test_shared_input_hessian_dedup.py \
  tests/test_gptq.py::test_embedding_capture_applies_rank_two_attention_mask
# 40 passed, 14 warnings in 5.93s
```

Both two-GPU regressions fail before the fix. No tests skipped; warnings are TorchScript deprecations. CUDA cases skip when the required devices are unavailable.

## Review Requirements

- [x] I personally reviewed every file in this diff.
- [x] I checked that the code matches existing project structure, APIs, and conventions.
- [x] I avoided unnecessary monkeypatching and used the project's normal extension points where possible.

### CODE REVIEW

**Review by Qubitium — APPROVED** [review]

**Comment by Qubitium** [comment]
@peter-positron Thank you! Confirmed fix locally.
